packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/data_sources/remote_data_sources/interceptors/analytics_interceptor.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2021, Prime Holding JSC
+// https://www.primeholding.com
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:dio/dio.dart';
+
+/// Interceptors are a simple way to intercept and modify http requests globally
+/// before they are sent to the server. That allows us to configure
+/// authentication tokens, add logs of the requests,
+/// add custom headers and much more.
+/// Interceptors can perform a variety of implicit tasks, from authentication
+/// to logging, for every HTTP request/response. Without interception, we will
+/// have to implement these tasks explicitly for each HttpClient method call.
+/// We use analytics interceptor to push notifications, present adds
+/// and collect data for users behaviour.
+/// You should implement one or more methods from the contract.
+class AnalyticsInterceptor extends Interceptor{
+
+  @override
+  void onRequest(RequestOptions options, RequestInterceptorHandler handler) {
+    // TODO: implement onRequest
+    super.onRequest(options, handler);
+  }
+
+  @override
+  void onResponse(Response response, ResponseInterceptorHandler handler) {
+    // TODO: implement onResponse
+    super.onResponse(response, handler);
+  }
+
+  @override
+  void onError(DioError err, ErrorInterceptorHandler handler) {
+    // TODO: implement onError
+    super.onError(err, handler);
```

Comment:
@RomanovaPrime, it would be great if we have `onError` implemented. In this way, we can track all API failures.

Reply:
I suppose I have to create a new instance of Dio in this class, in order to have opportunity to catch errors and retry to all APIs we use. So I tried to implement it in this way.